src/pages/admin/create-notes.tsx

```diff
@@ -0,0 +1,418 @@
+      notesList: [],
+      noteEdit: null,
+      noteEditObject: null,
+
+      // the power to delete things
+      delete: '',
+      understand: '',
+
+      // upload object
+      noteObject: {}
+    }
+
+    this.listNotes(null);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleDeleteNote = this.handleDeleteNote.bind(this);
+  }
+
+  listNotes(nextToken: any) {
+
+    var listNotess:any = API.graphql({
+        query: queries.listNotess,
+        variables: { nextToken: nextToken, sortDirection: "DESC", limit: 200 },
+        authMode: GRAPHQL_AUTH_MODE.API_KEY
+    });
+
+    listNotess.then((json: any) => {
+        console.log({ "Success customQueries.listNotess: ": json });
+        this.setState({
+              notesList: this.state.notesList.concat(json.data.listNotess.items).sort(function(a: any, b: any) {
+              var nameA = a.id.toUpperCase();
+              var nameB = b.id.toUpperCase();
+              if (nameA < nameB) {
+                return 1;
+              }
+              if (nameA > nameB) {
+                return -1;
+              }
+              return 0;
+            })
+        })
+        if (json.data.listNotess.nextToken != null)
+            this.listNotes(json.data.listNotess.nextToken)
+
+    }).catch((e: any) => { console.log(e) })
+  }
+
+  handleDeleteNote() {
+    if (this.state.understand === "Delete forever") {
+      var deleteNotes:any = API.graphql({
+          query: mutations.deleteNotes,
+          variables: { input: {'id': this.state.delete} },
+          authMode: GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS
+      });
+
+      deleteNotes.then((json: any) => {
+          console.log({ "Success mutations.deleteNotes: ": json });
+          this.setState({
+            delete: '',
+            understand: '',
+            showAlert: '⚠️ Deleted'
+          })
+
+      }).catch((e: any) => { console.log(e) })
+      return true;
+    } else {
+      this.setState({ showAlert: '⚠️ You must type the confirmation message' })
+    }
+  }
+
+  handleSave() {
+    if (this.state.editorState.getCurrentContent().hasText() === false || this.state.date === null) {
+      this.setState({ showAlert: "⚠️ You need a valid content and date to save." })
+      return false;
+    } else {
+      var contentState = this.state.editorState.getCurrentContent();
+      var raw = convertToRaw(contentState);
+      var html = draftToHtml(raw)
+      this.updateField('content', html)
+
+      console.log(this.state.noteObject)
+
+      this.updateField('tags', this.state.selectedTags)
+
+      this.updateField('title', this.state.title)
+      this.updateField('pdf', this.state.pdf)
+
+      var createNotes:any = API.graphql({
+          query: mutations.createNotes,
+          variables: { input: this.state.noteObject },
+          authMode: GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS
+      });
+
+      createNotes.then((json: any) => {
```

Comment:
Is there a reason you aren't using `await` here?

Reply:
Not really, this is just the syntax used throughout the code base. Is there a benefit to use `await` in this situation?

Reply:
Readability is one benefit. The other is that you unify all your error handling. Using async/await causes errors to be thrown in the normal synchronous way, so you can just use the normal
```
try {
    let resp = await API.graphql({ ... })
    let resp = await API.graphql({ ... })
    let resp = await API.graphql({ ... })
} catch (e) {
    console.error(e);
}
```
You can also just not `catch` and let the caller deal with the error, either as a Promise or using their own async/await. If you want all of your requests to run in parallel you can do
```
try {
    let resp = await Promise.all([
        API.graphql({ ... }),
        API.graphql({ ... }),
        API.graphql({ ... })
    ]);
} catch (e) {
    console.error(e);
}
```
I would recommend using `async`/`await` wherever possible, and slowly migrate the codebase to that pattern.